project/plugins.sbt

```diff
@@ -32,3 +32,5 @@ dependencyOverrides ++= Seq(
   "org.codehaus.plexus" % "plexus-utils" % "3.0.17",
   "com.google.guava" % "guava" % "20.0"
 )
+
+// addSbtPlugin("com.orrsella" % "sbt-stats" % "1.0.9-SNAPSHOT")
```

Comment:
Instead of doing a pull rquest to the sbt-stats plugin, what if instead of modifying the sbt-status plugin we create our own Analyzer that gathers the stats we care about (e.g. test vs src, tdml vs scala vs xml, etc.). Then we can configure the plugin in our build.sbt to use our custom analyzer. A bunch of the default stats (file size, chars, comments) seem kindof uninteresting to me. Setting own Analyzer would let us remove those and just output what we care about. Seems like an eaiser path and more configurable towards what we want than trying to upstream to sbt-stats.

Reply:
Can't be done. sbt-stats isn't flexible enough in the way it feeds files for processing to the analyzers. Basically, if sbt sources doesn't have the files you want stats on, you can't analyze them. So I modified it to pass separately to the analysis 4 things - sources, test sources, resources, test resources. That way analyzers can do what they want.

Reply:
Ah, that's a bummer. So it only provides the files in ``src/main/{java/scala}``? Maybe a more backwards compatible change and something more sbt-y would be to do something like:

1. Add a new ``analyzeResources()`` method, which would be called in additional to analyze, and would pass in resources for when people need to analyze stuff in ``src/main/resources/``. A default method could be provided to in the base class to return nothing so existing stuff doesn't break.
2. Make it so stats can run in the "test" configuration. Then you could run ``sbt test:stats`` to get stats about things in ``src/test/{java,scala}`` and ``src/test/resources``.'

These changes would make it so output doesn't change for anyone already using the plugin, but could be used for people like us that need status about resources and tests.

Reply:
I'm open to these ideas. 

They are unrelated to this PR which just deletes the old line-counter and suggests, via a comment in the plugins,  that sbt-stats be used.

The PR of changes to sbt-stats is:

https://github.com/orrsella/sbt-stats/pull/13

What is there and on openDFDL/sbt-stats computes the stats we've been incorporating into presentations about Daffodil, so meets our needs. But as you point out, there might be a more extensible way to do this change though that would maintain compatibility for existing sbt-stats users.

Reply:
Yeah, I would just prefer to not have to maintain a fork of sbt stats. If we could get something upstreamed that would be ideal. If they accept the current PR, that work for me.

Reply:
Not sure anything is being upstreamed. There's no activity for 2 years on sbt-stats.

Reply:
Ugh, I guess it doesn't matter. Seems fine to me.

Only thing I'd say is that the commented out ``addSbtPlugin`` line might be confusing since the "real" sbt-stats plugin doesn't have a version 1.0.9-SNAPSHOT. Thoughts on removing that line and we can provide documentation somewhere (maybe on the wiki?) on how to install and enable the custom line counter tailored towards counting lines in Daffodil and DFDL projects?

Something like:
1. CLone our fork from XYZ
2. Install it with ``sbt localInstall``
3. Modify ``~/.sbt/1.0/plugins/plugins.sbt`` to have the ``addSbtPlugin`` line
4. Run ``sbt stats`` on any project (Daffodil or DFDL schema project) to get the stats for that project.

+1

Reply:
Created https://cwiki.apache.org/confluence/display/DAFFODIL/Code+Stats+For+Reporting

Reply:
2 should actually be 'sbt publishLocal'. I've updated the wiki.